proxy/handler.go

```diff
@@ -9,14 +9,16 @@ import (
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/transport"
+	// "google.golang.org/grpc/transport"
+	// "google.golang.org/grpc/transport"
 )
 
 var (
 	clientStreamDescForProxying = &grpc.StreamDesc{
 		ServerStreams: true,
 		ClientStreams: true,
 	}
+	HandleEndCallback func(context.Context, *grpc.ClientConn)
```

Comment:
It would be better, IMO, to allow the stream director to handle connection management, rather than going through a global callback. This was implemented in #16, which is also available as a separate repo: https://github.com/vgough/grpc-proxy/blob/master/proxy/handler.go.